Attempt to improve description of security properties (#250)
oauth-wg/oauth-selective-disclosure-jwt: draft-ietf-oauth-selective-disclosure-jwt.md

Security considerations in this section help achieve the following properties:

**Selective Disclosure:** An adversary in the role of the Verifier cannot obtain
information from an SD-JWT about any claim name or claim value that was not
explicitly disclosed by the Holder unless that information can be derived from

> other disclosed claims or external sources.

Author: For example, if `nationalities` is selectively disclosable, and the holder does not disclose it to the verifier, the verifier might still learn the nationality from 

 - a passport image acquired out of band, e.g., during a video identification session (external source), or
 - a disclosed claim containing the passport number, which may leak information about the nationality ("other disclosed claims").

So we have to say "the verifier does not learn the `nationality` claim's value from the presentation of that claim" but it still may learn it from somewhere else because we cannot control that.

> explicitly disclosed by the Holder unless that information can be derived from
> other disclosed claims or external sources.

Reviewer: ```suggestion
explicitly disclosed by the Holder unless that information can be derived from
other disclosed claims or is available from sources other than SD-JWT.
```

Author: How about this?
```suggestion
explicitly disclosed by the Holder unless that information can be derived from
other disclosed claims or sources other than the presented SD-JWT.
```

**Integrity:** A malicious Holder cannot modify names or values of selectively disclosable claims without detection by the Verifier.

> Additionally, Holder Binding can be achieved as described in (#holder_binding_security).

Author: Maybe we should use the same format as for the other two properties. Re-using text from Brian's proposal:

```suggestion
**Holder Binding:** When Holder Binding is enforced, only the legitimate Holder of the credential can present the SD-JWT (as described in (#holder_binding_security)).
```

Reviewer: I would be ok with 
> **Holder Binding:** Only the legitimate Holder of the credential can present the SD-JWT (as described in (#holder_binding_security)).

Security considerations in this section help achieve the following properties:

> **Selective Disclosure:** An adversary in the role of the Verifier cannot obtain

Reviewer: ```suggestion
* An adversary in the role of the Verifier cannot obtain
```
I find two abstractions of a property achieved by security considerations confusing.

Author: Not sure what you mean by abstractions, but it is usually helpful to introduce names for the security properties you want to achieve. I'm open to suggestions for better names.

Reviewer: is "selective disclosure" a security property..?

Author: It is if we define it, and I think we should because that's the main goal of the spec. I'm open to better names, though.

information from an SD-JWT about any claim name or claim value that was not
explicitly disclosed by the Holder unless that information can be derived from
other disclosed claims or external sources.

> **Integrity:** The claims verified by a Verifier are a subset of the claims
> released by the Issuer, even if the Holder is malicious (i.e., a Holder cannot
> modify claim names or claim values without the Verifier noticing during verification).

Reviewer 2: I'm finding the main sentence kinda hard to process. What about just something inspired by the text you had in the parenthetical? 

```suggestion
**Integrity:** A malicious Holder cannot modify names or values of selectively disclosable claims without detection by the Verifier.
```

Reviewer: I like Brian's suggested text. Would also suggest removing **Integrity**. Per a comment above, I find two introduced abstractions confusing.

> Additionally, Holder Binding can be achieved as described in (#holder_binding_security).

Reviewer: Suggest removing this text. Holder binding ensures that it is the holder that directly received combined format for issuance that is taking to the verifier. But as Orie's PR suggests, some use-cases might want to allow it.

```suggestion
```

Author: I mention holder binding here as something you might want to achieve, as it is another important security property.

Reviewer: then it should be explained more, including a security property HB achieves

Reviewer 2: Maybe something like this then? Tries to explain it a bit while saying what it achieves.  

```suggestion
Additionally, as described in (#holder_binding_security), the application of Holder Binding can ensure that the presenter of an SD-JWT credential is the legitimate Holder of the credential.
```
